frontend/src/features/user/billing/BillCharges/BillCharges.tsx

```diff
@@ -0,0 +1,253 @@
+import { Dispatch, SetStateAction, useEffect, useMemo, useState } from 'react'
+import { useQuery } from 'react-query'
+import {
+  Box,
+  Container,
+  Flex,
+  Grid,
+  Skeleton,
+  Stack,
+  Text,
+} from '@chakra-ui/react'
+import simplur from 'simplur'
+
+import Pagination from '~/components/Pagination'
+
+import { SingleSelect } from '~components/Dropdown'
+import Searchbar from '~components/Searchbar'
+import Spinner from '~components/Spinner'
+
+import { getBillingInfo } from '~features/user/billing/BillingService'
+
+import { EsrvcIdFormInputs } from '../BillingForm'
+import { DateRange, dateRangeToString, stringToDateRange } from '../DateRange'
+
+import { BillingDownloadButton } from './components/BillingDownloadButton'
+import { BillingTable } from './components/BillingTable'
+
+export type BillChargesProps = {
+  esrvcId: string
+  dateRange: DateRange
+  todayDateRange: DateRange
+  setDateRange: Dispatch<SetStateAction<DateRange>>
+  onSubmitEsrvcId: (inputs: EsrvcIdFormInputs) => Promise<void>
+}
+
+const getSelectDateDropdownItems = ({ yr, mth }: DateRange): string[] => {
+  const selectDateDropdownItems: DateRange[] = []
+
+  // Insert this yr's months first
+  for (let loopMth = mth; loopMth >= 0; loopMth--) {
+    selectDateDropdownItems.push({ yr, mth: loopMth })
+  }
+
+  // Go backwards until 2019
+  for (let loopYr = yr - 1; loopYr >= 2019; loopYr--) {
+    for (let loopMth = 11; loopMth >= 0; loopMth--) {
+      selectDateDropdownItems.push({ yr: loopYr, mth: loopMth })
+    }
+  }
+  return selectDateDropdownItems.map(dateRangeToString)
+}
+
+export const BillCharges = ({
+  esrvcId,
+  dateRange,
+  todayDateRange,
+  setDateRange,
+  onSubmitEsrvcId,
+}: BillChargesProps): JSX.Element => {
+  const [currentPage, setCurrentPage] = useState<number>(1)
+  const [searchBarExpanded, setSearchBarExpanded] = useState<boolean>(false)
+
+  // Query for billing info
+  const {
+    data: { loginStats } = { loginStats: [] },
+    isLoading,
+    isRefetching,
+    refetch,
+  } = useQuery(esrvcId, () =>
+    getBillingInfo({
+      esrvcId,
+      yr: dateRange.yr.toString(),
+      mth: dateRange.mth.toString(),
+    }),
+  )
+
+  useEffect(() => {
+    refetch()
+  }, [refetch, dateRange])
+
+  // Compute some quantities for rendering
+
+  const statsCount = useMemo(() => loginStats.length, [loginStats])
+
+  const loginCount = useMemo(
+    () =>
+      loginStats
+        ?.map(({ total }) => {
+          return total
+        })
+        .reduce((acc, curr) => {
+          return acc + curr
+        }, 0) ?? 0,
```

Comment:
The `.map()` step is not needed.
```suggestion
        ?.reduce((acc, { total }) => {
          return acc + total
        }, 0) ?? 0,
```

Reply:
you're right! thanks for pointing that out :)